src/libasr/pass/python_bind.cpp

```diff
@@ -0,0 +1,406 @@
+#include <libasr/asr.h>
+#include <libasr/containers.h>
+#include <libasr/exception.h>
+#include <libasr/asr_utils.h>
+#include <libasr/asr_verify.h>
+#include <libasr/pass/pass_utils.h>
+#include <libasr/pass/replace_arr_slice.h>
+
+
+namespace LCompilers {
+ASR::expr_t *cpython_to_native(Allocator &al, ASR::expr_t *exp, ASR::ttype_t *type, const ASR::Function_t &f,
+                                const SymbolTable &parent_scope) {
+    ASR::ttype_t *i1_type = ASRUtils::TYPE(ASR::make_Integer_t(al, f.base.base.loc, 1));
+    ASR::ttype_t *i1ptr_type = ASRUtils::TYPE(ASR::make_Pointer_t(al, f.base.base.loc, i1_type));
+    ASR::ttype_t *i4_type = ASRUtils::TYPE(ASR::make_Integer_t(al, f.base.base.loc, 4));
+    ASR::ttype_t *i8_type = ASRUtils::TYPE(ASR::make_Integer_t(al, f.base.base.loc, 8));
+    ASR::ttype_t *u8_type = ASRUtils::TYPE(ASR::make_UnsignedInteger_t(al, f.base.base.loc, 8));
+    ASR::ttype_t *f8_type = ASRUtils::TYPE(ASR::make_Real_t(al, f.base.base.loc, 8));
+    ASR::ttype_t *ptr_t = ASRUtils::TYPE(ASR::make_CPtr_t(al, f.base.base.loc));
+
+    ASR::expr_t *conv_result = nullptr;
+    if (type->type == ASR::ttypeType::Integer) {
+        ASR::symbol_t *sym_PyLong_AsLongLong = parent_scope.get_symbol("PyLong_AsLongLong");
+        Vec<ASR::call_arg_t> args_PyLong_AsLongLong;
+        args_PyLong_AsLongLong.reserve(al, 1);
+        args_PyLong_AsLongLong.push_back(al, {f.base.base.loc, exp});
+        conv_result = ASRUtils::EXPR(ASR::make_Cast_t(al, f.base.base.loc,
```

Comment:
Why do we need to do `Cast_t` here (and the rest of the types below)? Isn't the returned value by the specific function (in this case `PyLong_AsLongLong`) already of the desired type?

Reply:
The signature of `PyLong_AsLongLong` is
`long long PyLong_AsLongLong(PyObject*)`. i.e. 
`i64 PyLong_AsLongLong(CPtr)`.

Python does not have `PyLongAsInt` that returns an `int` (`i32`).

The `Cast_t` is required in the following situation.
Assume the following PythonBind function
```
@pythoncall(module="mypylib")
def f(a: i32) -> i32:
    pass
```
or
```
@pythoncall(module="mypylib")
def f(a: i8) -> i8:
    pass
```
The `Cast_t` casts the result of `PyLong_AsLongLong` from `i64` to `i32` or `i8` according to the actual type required.

Reply:
Makes sense. Thanks.